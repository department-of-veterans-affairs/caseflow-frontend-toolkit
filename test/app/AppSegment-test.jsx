import React from 'react';
import AppSegment from '../../components/AppSegment';
import { render } from '@testing-library/react';

describe('AppSegment', () => {
  it('renders with no props', () => {
    const { asFragment } = render(
      <AppSegment>
        <p>children</p>
      </AppSegment>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders with filledBackground prop', () => {
    const { container, asFragment } = render(
      <AppSegment filledBackground>
        <p>children</p>
      </AppSegment>
    );

    expect(container.querySelector('.cf-app-segment--alt')).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders with noMarginTop prop', () => {
    const { asFragment } = render(
      <AppSegment noMarginTop>
        <p>children</p>
      </AppSegment>
    );
    // Since marginTop is applied via CSS-in-JS, the snapshot should capture the styling

    expect(asFragment()).toMatchSnapshot();
  });

  it('applies custom styling', () => {
    const { asFragment } = render(
      <AppSegment styling={{ color: 'red' }}>
        <p>children</p>
      </AppSegment>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
