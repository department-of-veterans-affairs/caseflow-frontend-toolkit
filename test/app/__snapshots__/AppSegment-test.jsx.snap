// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppSegment applies custom styling 1`] = `
<DocumentFragment>
  <section
    class="cf-app-segment"
    color="red"
  >
    <p>
      children
    </p>
  </section>
</DocumentFragment>
`;

exports[`AppSegment renders with filledBackground prop 1`] = `
<DocumentFragment>
  <section
    class="cf-app-segment cf-app-segment--alt"
  >
    <p>
      children
    </p>
  </section>
</DocumentFragment>
`;

exports[`AppSegment renders with no props 1`] = `
<DocumentFragment>
  <section
    class="cf-app-segment"
  >
    <p>
      children
    </p>
  </section>
</DocumentFragment>
`;

exports[`AppSegment renders with noMarginTop prop 1`] = `
<DocumentFragment>
  <section
    class="cf-app-segment"
    data-css-1pgs8gs=""
  >
    <p>
      children
    </p>
  </section>
</DocumentFragment>
`;
